src/state.rs

```diff
@@ -267,11 +261,29 @@ impl ProxyState {
                 debug!("failed to fetch workload for {}", ep.workload_uid);
                 return None;
             };
+
+            let target_port = svc.ports.get(&addr.port());
```

Comment:
Nit: can probably avoid 3x allocs to the same binding name with `or_else` or something.

A debug log would be nice if we mutate the port based on the tunnel setting.

Reply:
ended up using some if  {} else if {} so I could `return` (can't with the closures)